app/controllers/trainees/placements_controller.rb

```diff
@@ -26,9 +29,31 @@ def create
       end
     end
 
+    def update
+      @placement_form.update_placement(placement_params)
+
+      if @placement_form.save_or_stash
+        redirect_to(relevant_redirect_path)
+      else
+        render(:edit)
+      end
+    end
+
   private
 
-    def new_placement_params
+    def relevant_redirect_path
+      draft_apply_application? ? page_tracker.last_origin_page_path : trainee_placements_confirm_path(trainee)
+    end
+
+    def placements_form
+      @placements_form ||= PlacementsForm.new(trainee)
+    end
+
+    def set_placement_form
```

Comment:
could we memoise this too to be consistent eg `placement_form`? I don't think we use much `set_var` in this codebase

Reply:
Done https://github.com/DFE-Digital/register-trainee-teachers/pull/3749/commits/69ef28a654a484a814a61c631e0076dc4f979022